src/pages/playlist/playlist.js

```diff
@@ -0,0 +1,174 @@
+
+  //delete playlist
+  const deletePlaylist = (item) => {
+    deletePlaylistApi(
+      item._id,
+      playlistVideoDispatch,
+      setToggle,
+      toastDispatch
+    );
+  };
+
+  return (
+    <>
+      <div className={display ? "body-open-modal" : ""}>
+        <Navbar />
+        <section className="main-page playlist-page">
+          <Sidebar />
+          <div className="col2">
+            {/* <---------playlist Videos-------------> */}
+            {allPlaylistFromApi[0] === undefined ||
+            allPlaylistFromApi === null ? (
+              <EmptyPage text="You have not created any playlist" />
+            ) : (
+              <h1 className="playlistHeading">All Playlists</h1>
+            )}
+
+            {/* Create playlist icon ----------> */}
+            <div className="createPlaylist">
+              <button onClick={showDialog}> + Create Playlist</button>
+            </div>
+            {allPlaylistFromApi.map((item) => {
+              return (
+                <div className="playlist">
+                  <Link
+                    to="/videosInPlaylistPage"
+                    onClick={() => {
+                      setVideosInPlaylist(item.videos);
+                    }}
+                  >
+                    <div className="PlaylistContainer">
+                      <h1>{item.title}</h1>
+                    </div>
+                    <h1 className="play">
+                      <i className="fas fa-play"></i>play
+                    </h1>
+                    <div className="playlistIcon">
+                      <span
+                        className="iconify"
+                        data-icon="carbon:playlist"
+                      ></span>
+                    </div>
+                  </Link>
+
+                  <span
+                    onClick={() => deletePlaylist(item)}
+                    className="playlistDustbin"
+                  >
+                    <i className="fas fa-trash-alt fa-2x"></i>
+                  </span>
+                </div>
+              );
+            })}
+          </div>
+        </section>
+      </div>
+      {/*------------ Playlist Modal -----------------> */}
+      <div
+        className={display ? "modal dialog-box-true" : "modal dialog-box-false"}
+      >
+        <div className="modal-navbar">
+          <span onClick={hideDialog}>
+            <i className="fas fa-times"></i>
+          </span>
+        </div>
+        <div className="modal-footer flex-row"></div>
+        <div className="create-newPlaylist-wrapper mt-2">
+          <input
+            onChange={(e) => playlistTitleHandler(e)}
+            className="create-newPlaylist"
+            type="text"
+            id="newPlaylist"
+            name="newPlaylist"
+            placeholder="Enter new playlist name"
+            value={playlistNameValue}
+          />
+          <br />
+          <button
+            onClick={() => createNewPlaylistHandler(undefined, playlistTitle)}
```

Comment:
why passing undefined here?

Reply:
beacuse here i have not passed any video while creating the playlist but at some places when i create playlist by clicking on an individual video then i have passed the" video object" instead of "undefined".so,createApi function checks if its undefined or  not.If undefined then no need to add any video into the playlist if their is a "video object" then add that into playlist by calling the API function for adding the video into the playlist.